src/main/cpp/src/GpuTimeZoneDBJni.cpp

```diff
@@ -93,4 +93,28 @@ Java_com_nvidia_spark_rapids_jni_GpuTimeZoneDB_convertTimestampColumnToUTCWithTz
   }
   CATCH_STD(env, 0);
 }
+
+JNIEXPORT jlong JNICALL
+Java_com_nvidia_spark_rapids_jni_GpuTimeZoneDB_convertBetweenTimezones(JNIEnv* env,
+                                                                       jclass,
+                                                                       jlong input_handle,
+                                                                       jlong writer_tz_info_table,
+                                                                       jint writer_tz_raw_offset,
+                                                                       jlong reader_tz_info_table,
+                                                                       jint reader_tz_raw_offset)
+{
+  JNI_NULL_CHECK(env, input_handle, "input column is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    auto const input              = reinterpret_cast<cudf::column_view const*>(input_handle);
+    auto const writer_tz_info_tab = reinterpret_cast<cudf::table_view const*>(writer_tz_info_table);
+    auto const reader_tz_info_tab = reinterpret_cast<cudf::table_view const*>(reader_tz_info_table);
+    return cudf::jni::ptr_as_jlong(
+      spark_rapids_jni::convert_between_timezones(
+        *input, writer_tz_info_tab, writer_tz_raw_offset, reader_tz_info_tab, reader_tz_raw_offset)
+        .release());
+  }
+  CATCH_STD(env, 0);
```

Comment:
Please adopt the new convention `JNI_TRY`/`JNI_CATCH` as introduced in https://github.com/NVIDIA/spark-rapids-jni/pull/3714.